lib/tapyrus/tip0137.rb

```diff
@@ -0,0 +1,150 @@
+module Tapyrus
+  module TIP0137
+    # @param key [Tapyrus::Key] private key
+    # @param txid [String] txid
+    # @param index [Integer] index of transaction output
+    # @param color_id [Tapyrus::Color::ColorIdentifier] a valid color id
+    # @param value [Integer] non-negative integer less than 2^64
+    # @param script_pubkey [Tapyrus::Script] script pubkey in transaction output
+    # @param address [String] Tapyrus address for transaction output
+    # @param data [String] hexdecimal string
+    # @param client [Tapyrus::RPC::TapyrusCoreClient] rpc client
+    # @return [String] Signed message with JWS Format
+    # @raise [ArgumentError] raise if txid is not a 64-character hexadecimal string
+    # @raise [ArgumentError] raise if index is not a non-negative integer less than 2^32
+    # @raise [ArgumentError] raise if color_id is an invalid Tapyrus::Color::ColorIdentifier object
+    # @raise [ArgumentError] raise if value is not a non-negative integer less than 2^64
+    # @raise [ArgumentError] raise if script_pubkey is an invalid Tapyrus::Script object
+    # @raise [ArgumentError] raise if address is an invalid Tapyrus address
+    # @raise [ArgumentError] raise if data is not a hexdecimal string
+    # @raise [ArgumentError] raise if transaction is not found with txid and index in blockchain 
+    # @raise [ArgumentError] raise if script and value is not equal to ones in blockchain
+    def sign_message!(
+      key,
+      txid:,
+      index:,
+      value: ,
+      script_pubkey:,
+      color_id: nil,
+      address: nil,
+      data: nil,
```

Comment:
This `data` refer to the message being signed? If that's the case, I think it would be better to name it `message` rather than `data` to be more consistent with the method name.